src/rust/engine/process_execution/src/local.rs

```diff
@@ -135,8 +149,16 @@ impl super::CommandRunner for CommandRunner {
             )
           })
           // Force workdir not to get dropped until after we've ingested the outputs
-          .map(|result| (result, workdir))
-          .map(|(result, _workdir)| result)
+          .map(move |result| (result, workdir, cleanup_local_dirs) )
+          .map(move |(result, workdir, cleanup_local_dirs)| {
+            if cleanup_local_dirs == false {
+              // This consumes the `TempDir` without deleting directory on the filesystem, meaning
+              // that the temporary directory will no longer be automatically deleted when dropped.
+              let preserved_path = workdir.into_path();
+              info!("preserved local process execution dir: {:?}", preserved_path.into_os_string());
```

Comment:
`.into_os_string()` should be unnecessary here; `Path` implements `Debug`. (Or you can call `presersved_path.display()` and format with `{}` if you prefer)

Reply:
`Path` does not implement `Display` (for unicodey reasons), but os_string does on Unix. So a third option is: ```
info!("preserved local process execution dir: {}", preserved_path.as_os_str());
```

Reply:
done